From: Simon McVittie <smcv@debian.org>
Date: Thu, 19 Jan 2017 14:47:14 +0000
Subject: flatpak-system-helper: remove dangling reference to
 EXTERNAL_INSTALL_DIR

If EXTERNAL_INSTALL_DIR wasn't defined (which is likely, since nothing
in the build system does), then we would tell the system helper
to search /exports/share, which is not Flatpak's territory
(but is a semi-commonly-used path for site-specific NFS shares,
which might have contents that are inappropriate to search here).

Installations that do define an external installation directory
(Endless OS?) can easily override the Environment to include it
by placing a drop-in in
/etc/systemd/user/flatpak-system-helper.service.d/external.conf
or similar.

Forwarded: https://github.com/flatpak/flatpak/pull/497
---
 Makefile.am                                    | 1 -
 system-helper/flatpak-system-helper.service.in | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 24b2528..828e34e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -25,7 +25,6 @@ SUBDIRS += po
 %.service: %.service.in config.log
 	$(AM_V_GEN) $(SED) -e "s|\@libexecdir\@|$(libexecdir)|" \
 		-e "s|\@localstatedir\@|$(localstatedir)|" \
-		-e "s|\@externalinstalldir\@|$(EXTERNAL_INSTALL_DIR)|" \
 		-e "s|\@extraargs\@||" $< > $@
 
 dbus_servicedir = $(DBUS_SERVICE_DIR)
diff --git a/system-helper/flatpak-system-helper.service.in b/system-helper/flatpak-system-helper.service.in
index 7b4fc8c..2ba5e03 100644
--- a/system-helper/flatpak-system-helper.service.in
+++ b/system-helper/flatpak-system-helper.service.in
@@ -3,6 +3,6 @@ Description=flatpak system helper
 
 [Service]
 BusName=org.freedesktop.Flatpak.SystemHelper
-Environment=XDG_DATA_DIRS=@localstatedir@/lib/flatpak/exports/share/:@externalinstalldir@/exports/share/:/usr/local/share/:/usr/share/
+Environment=XDG_DATA_DIRS=@localstatedir@/lib/flatpak/exports/share/:/usr/local/share/:/usr/share/
 ExecStart=@libexecdir@/flatpak-system-helper
 Type=dbus
